syntax = "proto3";

service XRFAnalyzerService {
  rpc FindPeaksMessage (FindPeaksRequest) returns (FindPeaksReply);
  rpc BackgroundMessage (BackgroundRequest) returns (BackgroundReply);
}

message FindPeaksRequest {
  repeated double counts = 1;
  int32 height = 2;
  int32 threshold = 3;
  int32 distance = 4;
  int32 prominence = 5;
  int32 width = 6;
  int32 wlen = 7;
  double rel_height = 8;
  int32 plateau_size = 9;
}

message FindPeaksReply {
  repeated int32 peaks = 1;
  repeated int32 left_bases = 2;
  repeated int32 right_bases = 3;
}

message BackgroundRequest {
	repeated double counts = 1;
	int32 lambda_ = 2;
	int32 iterations = 3;
}

message BackgroundReply {
	repeated double corrected_counts = 1;
}